Misc - Add - IFAK, AFAK, MFAK

MissHeda opened the pull request:
**When merged this pull request will:**
- Add new items: IFAK, AFAK, MFAK
![image](https://github.com/KAT-Advanced-Medical/KAM/assets/95135995/edc8107f-501d-4215-9216-f100f70c15e0)

- Add according addon settings to configure everything to your liking.
![image](https://github.com/KAT-Advanced-Medical/KAM/assets/95135995/54633cc4-c8f7-4852-80c0-a1f5f68585e5)

- Add individual unpack & repack options to not unpack the whole FAK.
![image](https://github.com/KAT-Advanced-Medical/KAM/assets/95135995/6ee4e21e-b1a4-4e22-972a-42c788a679a0)
![image](https://github.com/KAT-Advanced-Medical/KAM/assets/95135995/d1d4f15c-3c5e-4c88-a035-694c41538d70)



- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

MissHeda committed:
First steps

MissHeda committed:
Second push

MissHeda committed:
Everything working

MissHeda committed:
downdate project.tomel

MissHeda committed:
validate Aghhhhh

MissHeda committed:
Updated version with correct names

MissHeda committed:
push to save new stuff

MissHeda committed:
IFAK Ready

MissHeda committed:
Changes before rework.

MissHeda committed:
Merge branch 'IFAK,-AFAK-&-MFAK' of https://github.com/MissHeda/KAM into IFAK,-AFAK-&-MFAK

MissHeda committed:
push

MissHeda committed:
update

MissHeda committed:
update check fnc

MissHeda committed:
Merge branch 'IFAK,-AFAK-&-MFAK' of https://github.com/MissHeda/KAM into IFAK,-AFAK-&-MFAK

MissHeda committed:
add contruct for unpackIFAK

Todo:
- redo CBA settings (MFAK not set at all, A & IFAK working for the moment)
- fill in whole unpack fnc
- add repack handeling & fnc.

MissHeda committed:
unpack IFAK file update

MissHeda committed:
Unpack fnc completly working

MissHeda committed:
CBA settings update

MissHeda committed:
Merge branch 'IFAK,-AFAK-&-MFAK' of https://github.com/MissHeda/KAM into IFAK,-AFAK-&-MFAK

MissHeda committed:
Add FAK repack check FNC & update Images & strings.

MissHeda committed:
final push

everything working.

MissHeda committed:
validate fix

AtrixZockt commented:
First things first, where do the icons come from? do we have permission to use them?

MissHeda commented:
All images where altered and fit under the Creative Commons licence.

AtrixZockt commented on addons/misc/functions/fnc_unpackFAK.sqf:
```diff
@@ -0,0 +1,6979 @@
+
+switch (_slotsToGive) do {
+    
+    case 0: { 
```
please add some comments on what case state is for what

MissHeda replied:
Pretty self explanatory if you ask me.
Slot 0 =  unpack the rest of all slots left. ( I give you that one, this one was hard to guess)
Slot 1= unpack slot 1
Slot 2 = unpack slot 2
-||- (and so on...)

BlueTheKing commented on addons/misc/functions/fnc_checkFAKSlot.sqf:
```diff
@@ -0,0 +1,3000 @@
+                        case Slot_1_3_4_5_6_8: { _hasTrueValue pushback true; };
+                        case Slot_1_3_4_5_7_8: { _hasTrueValue pushback true; };
+                        case Slot_1_3_4_6_7_8: { _hasTrueValue pushback true; };
+                        case Slot_1_3_5_6_7_8: { _hasTrueValue pushback true; };
+                        case Slot_1_4_5_6_7_8: { _hasTrueValue pushback true; };
+                    };
+                };
+
+                if (_x >= SlotsLeft_6D_3Min && _x <= SlotsLeft_6D_3Max) then {
+                    switch (_x) do {
+                        case Slot_2_3_4_5_6_7: { _hasTrueValue pushback false; };
+                        case Slot_2_3_4_5_6_8: { _hasTrueValue pushback true; };
+                        case Slot_2_3_4_5_7_8: { _hasTrueValue pushback true; };
+                        case Slot_2_3_4_6_7_8: { _hasTrueValue pushback true; };
+                        case Slot_2_3_5_6_7_8: { _hasTrueValue pushback true; };
+                        case Slot_2_4_5_6_7_8: { _hasTrueValue pushback true; };
+                        case Slot_3_4_5_6_7_8: { _hasTrueValue pushback true; };
+                    };
+                };
+            };
+
+            // 7 Slots left
+            if (_x >= SlotsLeft_7D_Min && _x <= SlotsLeft_7D_Max) then { 
+
+                switch (_x) do {
+                    case Slot_1_2_3_4_5_6_7: { _hasTrueValue pushback false; };
+                    case Slot_1_2_3_4_5_6_8: { _hasTrueValue pushback true; };
+                    case Slot_1_2_3_4_5_7_8: { _hasTrueValue pushback true; };
+                    case Slot_1_2_3_4_6_7_8: { _hasTrueValue pushback true; };
+                    case Slot_1_2_3_5_6_7_8: { _hasTrueValue pushback true; };
+                    case Slot_1_2_4_5_6_7_8: { _hasTrueValue pushback true; };
+                    case Slot_1_3_4_5_6_7_8: { _hasTrueValue pushback true; };
+                    case Slot_2_3_4_5_6_7_8: { _hasTrueValue pushback true; };
+                    case Slot_1_2_3_4_5_6_7_8: { _hasTrueValue pushback true; };
+                };
+            };
+        };
+    };
+
+} forEach _ammoCount;
```
Don't copy paste
Redo

MissHeda replied:
It's not copy past!
Each value is customly set for value.

BlueTheKing commented on addons/misc/functions/fnc_repackFAK.sqf:
```diff
@@ -0,0 +1,4902 @@
+            };
+
+            if (_ammoCount >= SlotsLeft_2D_2Min && _ammoCount <= SlotsLeft_2D_2Max) exitWith {
+                switch (_
```
Don't copy paste, redo

MissHeda replied:
It's not copy past!
Each value is customly set for value.

BlueTheKing commented on addons/misc/functions/fnc_unpackFAK.sqf:
```diff
@@ -0,0 +1,6979 @@
+                    [_unit, _fourthSlotItem_4, _container] call FUNC(arrayToInvItem);
+                };
+                case Slot_1_4: { 
+                    [_unit, _mag, _removeContainer, _ammoCount, Slo
```
Don't copy paste
Redo

MissHeda replied:
It's not copy past!
Each value is customly set for value.

BlueTheKing commented on addons/misc/stringtable.xml:
```diff
@@ -853,6 +853,178 @@
+            <German>Achte Tasche Gegenstand</German>
+        </Key>
+        <Key ID="STR_KAT_Misc_IFAK_Display">
+            <English>Individuell First Aid Kit</English>
```
Individual

BlueTheKing commented on addons/misc/CfgVehicles.hpp:
```diff
@@ -433,6 +433,546 @@ class CfgVehicles {
+
+                class Unpack_kat_IFAK_Item {
+                    displayName = CSTRING(IFAK_Unpack);
+                    condition = QUOTE([ARR_3(_player, ""kat_IFAK"", 0)] call FUNC(checkFAKSlot));
```
Use single quotes
```suggestion
                    condition = QUOTE([ARR_3(_player, 'kat_IFAK', 0)] call FUNC(checkFAKSlot));
```

BlueTheKing commented on addons/misc/CfgVehicles.hpp:
```diff
@@ -433,6 +433,546 @@ class CfgVehicles {
+                        statement = QUOTE([ARR_3(_player, ""kat_IFAK"", 2)] call FUNC(unpackFAK));
+                        showDisabled = 0;
+                        icon = QPATHTOF(ui\IFAK.paa);
+                    };
```
Inherit entries to avoid copy pasting
```suggestion
                    class Slot2: Slot1 {
                        displayName = CSTRING(FAK_Slot_2);
                        condition = QUOTE([ARR_3(_player, 'kat_IFAK', 2)] call FUNC(checkFAKSlot));
                        statement = QUOTE([ARR_3(_player, 'kat_IFAK', 2)] call FUNC(unpackFAK));
                    };
```

MissHeda committed:
requested changes

BlueTheKing committed:
Rewrite FAK functions

BlueTheKing committed:
Merge branch 'IFAK,-AFAK-&-MFAK' of https://github.com/MissHeda/KAM into pr/417

BlueTheKing committed:
Handle empty FAK

MissHeda committed:
updating string table

MissHeda committed:
Adjusting weight & items

First Slot = [M]assive Bleeding
Second Slot = [B]andages
Third Slot = [P]ainkiller / pain reduction
Fourth Slot = [P]neumotorax management
Fifth Slot = [A]irway management
Sixth Slot = [A]dvanced medication
Seventh Slot = [F]luid
Eighth Slot = [T]ools

AtrixZockt committed:
Merge branch 'dev-Tomcat' into IFAK,-AFAK-&-MFAK

AtrixZockt committed:
Merge branch 'dev-Tomcat' into IFAK,-AFAK-&-MFAK

MiszczuZPolski committed:
update path

MiszczuZPolski committed:
fix padding
